app/code/Magento/User/Controller/Adminhtml/Auth/Forgotpassword.php

```diff
@@ -7,26 +7,38 @@
 namespace Magento\User\Controller\Adminhtml\Auth;
 
 use Magento\Security\Model\SecurityManager;
+use Magento\Framework\App\ObjectManager;
 
 class Forgotpassword extends \Magento\User\Controller\Adminhtml\Auth
 {
     /**
      * @var SecurityManager
      */
     protected $securityManager;
+    
+    /**
+     * User model factory
+     *
+     * @var \Magento\User\Model\ResourceModel\User\CollectionFactory
+     */
+    private $userCollectionFactory;
 
     /**
      * @param \Magento\Backend\App\Action\Context $context
      * @param \Magento\User\Model\UserFactory $userFactory
      * @param \Magento\Security\Model\SecurityManager $securityManager
+     * @param \Magento\User\Model\ResourceModel\User\CollectionFactory $userCollectionFactory
      */
     public function __construct(
         \Magento\Backend\App\Action\Context $context,
         \Magento\User\Model\UserFactory $userFactory,
-        \Magento\Security\Model\SecurityManager $securityManager
+        \Magento\Security\Model\SecurityManager $securityManager,
+        \Magento\User\Model\ResourceModel\User\CollectionFactory $userCollectionFactory = null
```

Comment:
Maybe you could do the PR on 2.3-develop instead for the refactoring. Than you can remove the = null part and you don't need the if statement on line 42 and 43.

Reply:
I can do a forwardport for 2.3-develop.